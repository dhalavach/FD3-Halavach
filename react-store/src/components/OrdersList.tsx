import { useEffect, useState } from 'react';
import { orderObject } from '../types/Types';
import axios from 'axios';
import OrderListContainer from './OrderListContainer';

export default function OrdersList() {
  const [orders, setOrders] = useState<orderObject[]>([]);
  const [ordersLoaded, setOrdersLoaded] = useState(false);

  const fetchConfig = {
    PRODUCTS_URL: 'http://localhost:3000/products',
    ORDERS_URL: 'http://localhost:3000/orders',
    method: 'GET',
    headers: {
      Accept: 'application/json',
    },
  };
  const loadOrders = async () => {
    try {
      const response = await axios.get(fetchConfig.ORDERS_URL);
      setOrders(response.data);
      setOrdersLoaded(true);
    } catch (error) {
      console.error(error);
    }
  };

  useEffect(() => {
    loadOrders();
  }, []);

  return (
    <>
      <h3 className='orders-list__heading'>List of orders</h3>
      {!ordersLoaded
        ? 'Loading...'
        : orders.map((order: any) => {
            return <OrderListContainer order={order} key={order.id} />;
          })}
    </>
  );
}
