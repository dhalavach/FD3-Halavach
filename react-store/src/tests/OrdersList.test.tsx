import '@testing-library/jest-dom';
import OrdersList from '../components/OrdersList';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../components/Store';

test('component renders without throwing an error', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <OrdersList />
      </Provider>
    )
    .toJSON();
  expect(tree).toBeTruthy();
});

test('component matches the snapshot', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <OrdersList />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
