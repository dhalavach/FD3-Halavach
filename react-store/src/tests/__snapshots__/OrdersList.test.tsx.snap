// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component matches the snapshot 1`] = `
[
  <h3
    className="orders-list__heading"
  >
    List of orders
  </h3>,
  "Loading...",
]
`;
